woocommerce/woocommerce-gutenberg-products-block
# Reviews by category block

This is the reviews by category block, based on the products block. It supports showing reviews from **multiple** categories which is why I dropped the use of `withCategory` hoc.

The API has been updated to return product names and permalinks.

Edit view:

![Screenshot 2019-08-16 at 16 21 54](https://user-images.githubusercontent.com/90977/63178450-fb1e8780-c041-11e9-82f7-668059f7b4a3.png)

Frontend view:

![Screenshot 2019-08-16 at 16 18 58](https://user-images.githubusercontent.com/90977/63178269-995e1d80-c041-11e9-8e83-f81bbcd8d7ff.png)

Block settings:

![Screenshot 2019-08-16 at 16 19 52](https://user-images.githubusercontent.com/90977/63178311-b397fb80-c041-11e9-83c6-210c657c1c12.png)

Block settings toggle content like this, and if all are disabled a placeholder is shown:

![2019-08-16 16 20 33](https://user-images.githubusercontent.com/90977/63178383-d6c2ab00-c041-11e9-817b-976a0a7d917e.gif)

**To test**

- Create reviews by category block
- Toggle various cats on and off and test results
- Toggle visibility of all review parts and test results
- Save, ensure settings are preserved.
- Try filtering/viewing on the frontend.
- Confirm product permalinks correctly link to product.
- Confirm product reviews block still functions since some shared components were modified.

Closes #589
